### PR TITLE
Wdt 412

### PR DESCRIPTION
**Pull request by jshum2479**
This restore the return code of restart required if wlst isRestartRequired returns true during update operation.

### CODE REVIEW

File: core/src/main/python/wlsdeploy/util/cla_utils.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ class CommandLineArgUtil(object):
     USAGE_ERROR_EXIT_CODE          = 99
     ARG_VALIDATION_ERROR_EXIT_CODE = 98
     PROG_RESTART_REQUIRED          = 103
-    PROG_ROLLBACK_IF_RESTART_EXIT_CODE = 3
+    PROG_ROLLBACK_IF_RESTART_EXIT_CODE = 104
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/util/cla_utils.py:105]: Needs a message in shared.sh, shared.cmd in checkExitCode for this exit code

Comment (jshum2479) [reply core/src/main/python/wlsdeploy/util/cla_utils.py:105]: fixed

File: core/src/main/python/update.py [hunk 2]
```diff
@@ -144,15 +144,14 @@ def __update_online(model, model_context, aliases):
         __release_edit_session_and_disconnect()
         raise de
 
-    exit_code = 0
+    topology_updater.set_server_groups()
```
Comment (CarolynRountree) [inline core/src/main/python/update.py:147]: maybe move this between update() and deployResources()

Comment (ddsharpe) [reply core/src/main/python/update.py:147]: This change does not seem related to the rest of the changes.  Why is it necessary to add the set_server_groups here?

Comment (CarolynRountree) [reply core/src/main/python/update.py:147]: Never mind, I need to be in this code. I don't believe your code will work with extension templates by removing the edit, start edit. But my changes will remove the 3 statements.

Comment (jshum2479) [reply core/src/main/python/update.py:147]: It's not really adding the logic set_server_groups now, it was there before and just move it earlier when it is still in edit session.

File: core/src/main/python/wlsdeploy/util/cla_utils.py (hunk 1 above)
Comment (CarolynRountree) [inline core/src/main/python/wlsdeploy/util/cla_utils.py:105]: Just read the update document trying to find the command line argument to rollback. That argument isn't documented and also there are bad RCs in the documentation that need to be removed. 104 needs to be added.

```

    101 - The domain needs to be restarted and the Update Domain Tool needs to be re-invoked with the same arguments.
    102 - The servers impacted by the update operation need to be restarted, in a rolling fashion, starting with the Administration Server, if applicable.
    103 - The entire domain needs to be restarted.
```

Comment (jshum2479) [reply core/src/main/python/wlsdeploy/util/cla_utils.py:105]: Added

**Comment by jshum2479** [comment]
Hmm,  what's your test case looks like? and any logs?

**Comment by CarolynRountree** [comment]
1. created a domain with clusterslimit.yml and clusters.properties with following command line:  
```
weblogic-deploy\bin\createDomain.cmd -oracle_home c:\oracle\fmw12213 -domain_home c:\oracle\fmw12213\user_projects\domains\rollback_domain -model_file .\clusterslimits.yml -variable_file .\clusters.properties -domain_type JRF -run_rcu -rcu_prefix CERJRF -rcu_db localhost:32769/ORCLPDB1.localdomain
```
2. Started admin server
3. Ran update domain with clustersupdate.yml and this command line
```
weblogic-deploy\bin\updateDomain.cmd -oracle_home c:\oracle\fmw12213 -domain_home c:\oracle\fmw12213\user_projects\domains\rollback_domain -model_file .\clustersupdate.yml -domain_type JRF -rollback_if_require_restart -admin_url t3://localhost:7001 -admin_user weblogic
```

[clustersupdate.yml.txt](https://github.com/oracle/weblogic-deploy-tooling/files/4541882/clustersupdate.yml.txt)

[createDomain.log.txt](https://github.com/oracle/weblogic-deploy-tooling/files/4541890/createDomain.log.txt)

File: site/update.md [hunk 3]
```diff
@@ -23,6 +23,7 @@ When running the tool in WLST online mode, the update operation may require serv
 - `101` - The domain needs to be restarted and the Update Domain Tool needs to be re-invoked with the same arguments.
 - `102` - The servers impacted by the update operation need to be restarted, in a rolling fashion, starting with the Administration Server, if applicable.
```
Comment (CarolynRountree) [inline site/update.md:24]: the rollback command arg is not documented. And the 101, 102 are not valid are they?

File: site/deploy.md [hunk 4]
```diff
@@ -28,6 +28,7 @@ When running the tool in WLST online mode, the deploy operation may require serv
 - `101` - The domain needs to be restarted and the Deploy Applications Tool needs to be re-invoked with the same arguments.
```
Comment (CarolynRountree) [inline site/deploy.md:28]: 101 and 102 are not valid are they? the rollback command line arg is not documented.

File: site/update.md (hunk 3 above)
Comment (jshum2479) [reply site/update.md:24]: The rollback command arg is for future operator use only originally, let me add the doc in it. I didn't change 101 and 102, no idea where they come from

File: site/deploy.md (hunk 4 above)
Comment (jshum2479) [reply site/deploy.md]: I don't see anywhere returning 101, 102 from the source. I am removing them now.

**Comment by jshum2479** [comment]
Added online system test also

**Pull request merged**